docs/source/conf.py

```diff
@@ -92,15 +93,43 @@
 }
 
 # list of all labscript suite components that have docs
```

Comment:
Include this in a yaml file, whether or not adopting the suggestion to use a templating engine (below).

Reply:
I don't think it's worth putting this in a yaml file just to read it back into this one place. Especially since some data depends on a Python variable.

Reply:
It simplifies things quite a bit, especially if the top level is component type (libs/guis). See rpanderson/labscript-suite@91fdaad

The only variable it depended on is `img_path`, which I've specified in the template (less repetition, too).

Reply:
For example, this:
```python
template.render(
    libs=dict(
        filter(
            lambda x: x[1]['type'] == 'lib',
            labscript_suite_programs.items(),
        )
    ),
    guis=dict(
        filter(
            lambda x: x[1]['type'] == 'gui',
            labscript_suite_programs.items(),
        )
    ),
    toctree_entries=list(
        filter(lambda x: x != project, sorted(labscript_suite_programs))
    ),
    ...
)
```

becomes:
```python
component_names = [name for items in components.values() for name in items]
toctree_entries = sorted([x for x in component_names if x != project])
template.render(
    libs=components['libs'],
    guis=components['guis'],
    toctree_entries=toctree_entries
    ...
)
```

Reply:
So the `template.render` code can be simplified even more if the logic is just moved into the jinja template (which I've now done). I still don't see the appeal for moving the `labscript_suite_programs` dictionary into a separate file. It's just splitting the sphinx configuration across two files and two file formats.

Reply:
> I still don't see the appeal for moving the labscript_suite_programs dictionary into a separate file. It's just splitting the sphinx configuration across two files and two file formats.

The point is separating data and code. Yes, everything is data, but when the dictionary in question:
* Has a 39 line definition;
* Extends the length of the Python file by 20%; and
* Is amendable to a more concise definition (even when adding an additional level of hierarchical structure which simplifies the subsequent logic) in e.g. YAML,
it's worth considering.

> the `template.render` code can be simplified even more if the logic is just moved into the jinja template (which I've now done).

Cool simplification. This is one less reason to use the nested YAML format I proposed. `img_path` in the `labscript_suite_programs` dictionary is still redundant.

Reply:
> `img_path` in the `labscript_suite_programs` dictionary is still redundant.

This is now resolved